### PR TITLE
Serialization verifications

### LINKED ISSUE #77: Setting serializer values
https://github.com/pathfinder-for-autonomous-navigation/FlightSoftware/blob/26e79d950b53c2e63110d0053515d51f7b256989/src/FCCode/GomspaceController.cpp#L14-L25

### PR DESCRIPTION
**Pull request by tanishqaggarwal**
Closes #77 . Some additional issues were added to capture verifications that can't be settled immediately.

### Testing
Deferred. There's no real testing that can be done yet; these are configuration values that will affect us when we do full-mission testing.

### Constants
Some constants were moved around. One constant (computed from other constants) was added that is of general use: `orbit_ccno`.

### Telemetry
It's all telemetry config changes.

### Documentation Evidence
Either inline constants or PR comments should be added to document rationale for serializer constant choices.

### CODE REVIEW

File: src/fsw/FCCode/MissionManager.cpp [hunk 1]
```diff
@@ -16,21 +16,19 @@ const constexpr std::array<mission_state_t, 5> MissionManager::fault_responsive_
 const constexpr std::array<mission_state_t, 7> MissionManager::fault_nonresponsive_states;
 
 MissionManager::MissionManager(StateFieldRegistry &registry, unsigned int offset) : TimedControlTask<void>(registry, "mission_ct", offset),
-                                                                                    detumble_safety_factor_f("detumble_safety_factor", Serializer<double>(0, 1, 10)),
-                                                                                    close_approach_trigger_dist_f("trigger_dist.close_approach", Serializer<double>(0, 10000, 14)),
-                                                                                    docking_trigger_dist_f("trigger_dist.docking", Serializer<double>(0, 100, 10)),
+                                                                                    detumble_safety_factor_f("detumble_safety_factor", Serializer<double>(0, 1, 7)),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/MissionManager.cpp:19]: Allowing ~1% resolution in specifying a safety factor.

File: src/fsw/FCCode/MissionManager.cpp [hunk 2]
```diff
@@ -16,21 +16,19 @@ const constexpr std::array<mission_state_t, 5> MissionManager::fault_responsive_
...
+                                                                                    close_approach_trigger_dist_f("trigger_dist.close_approach", Serializer<double>(0, 5000, 13)),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/MissionManager.cpp:20]: Assuming a distance of ~5 km and allowing resolution for triggering this distance up to 1m.

File: src/fsw/FCCode/MissionManager.cpp [hunk 3]
```diff
@@ -16,21 +16,19 @@ const constexpr std::array<mission_state_t, 5> MissionManager::fault_responsive_
...
+                                                                                    docking_trigger_dist_f("trigger_dist.docking", Serializer<double>(0, 100, 14)),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/MissionManager.cpp:21]: Give threshold around 1mm resolution.

File: src/fsw/FCCode/MissionManager.cpp [hunk 4]
```diff
@@ -16,21 +16,19 @@ const constexpr std::array<mission_state_t, 5> MissionManager::fault_responsive_
...
                                                                                     docking_timeout_limit_f("docking_timeout_limit",
-                                                                                                            Serializer<unsigned int>(2 * PAN::one_day_ccno)),
-                                                                                    adcs_state_f("adcs.state", Serializer<unsigned char>(10)),
+                                                                                                            Serializer<unsigned int>(0, 2 * PAN::one_day_ccno, 6)),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/MissionManager.cpp:23]: Allow docking timeout to be set in units of 1 hour.

File: src/fsw/FCCode/MissionManager.cpp [hunk 5]
```diff
@@ -16,21 +16,19 @@ const constexpr std::array<mission_state_t, 5> MissionManager::fault_responsive_
...
                                                                                     docking_config_cmd_f("docksys.config_cmd", Serializer<bool>()),
                                                                                     enter_docking_cycle_f("docksys.enter_docking"),
                                                                                     mission_state_f("pan.state", Serializer<unsigned char>(12), 10),
                                                                                     is_deployed_f("pan.deployed", Serializer<bool>(), 1000),
-                                                                                    deployment_wait_elapsed_f("pan.deployment.elapsed", Serializer<unsigned int>(0, 15000, 32), 500),
-                                                                                    sat_designation_f("pan.sat_designation", Serializer<unsigned char>(2), 100),
+                                                                                    deployment_wait_elapsed_f("pan.deployment.elapsed", Serializer<unsigned int>(15000), 500),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/MissionManager.cpp:29]: This is not a field that would change after the deployment period; there's no need to downlink it, so no need to set good serialization parameters for it.

File: src/fsw/FCCode/PiksiControlTask.cpp [hunk 6]
```diff
@@ -7,10 +7,10 @@ PiksiControlTask::PiksiControlTask(StateFieldRegistry &registry,
     unsigned int offset, Devices::Piksi &_piksi) 
     : TimedControlTask<void>(registry, "piksi", offset),
     piksi(_piksi),
-    pos_f("piksi.pos", Serializer<d_vector_t>(0,100000,100)),
-    vel_f("piksi.vel", Serializer<d_vector_t>(0,100000,100)),
-    baseline_pos_f("piksi.baseline_pos", Serializer<d_vector_t>(0,100000,100)),
-    current_state_f("piksi.state", Serializer<unsigned char>(10)),
+    pos_f("piksi.pos", Serializer<d_vector_t>(6771000,6971000, 28 + SerializerConstants::min_dvsz)),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/PiksiControlTask.cpp:10]: See comments in FieldCreatorTask for rationale for these serialization values.

File: src/fsw/FCCode/TimedControlTask.hpp [hunk 7]
```diff
@@ -178,7 +178,8 @@ class TimedControlTask : public ControlTask<T>, public TimedControlTaskBase {
         num_lates_field_name("timing." + name + ".num_lates"),
         num_lates_f(num_lates_field_name, Serializer<unsigned int>()),
         avg_wait_field_name("timing." + name + ".avg_wait"),
-        avg_wait_f(avg_wait_field_name, Serializer<float>(0,PAN::control_cycle_time_us,32))
+
+        avg_wait_f(avg_wait_field_name, Serializer<float>(0,PAN::control_cycle_time_us, 18))
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/TimedControlTask.hpp:182]: This field will almost never be downlinked, but this change reflects the fact that the average wait should be discretized on the order of 1 microsecond.

File: src/fsw/FCCode/FieldCreatorTask.hpp [hunk 8]
```diff
@@ -20,17 +20,15 @@ class FieldCreatorTask : public ControlTask<void> {
...
 
       FieldCreatorTask(StateFieldRegistry& r) : 
         ControlTask<void>(r),
         time_f("orbit.time", Serializer<double>(0.0, 18'446'744'073'709'551'616.0, 64)),
-        pos_f("orbit.pos", Serializer<lin::Vector3d>(0,100000,100)),
-        vel_f("orbit.vel", Serializer<lin::Vector3d>(0,100000,100)),
-        pos_baseline_f("orbit.baseline_pos", Serializer<lin::Vector3d>(0,100000,100)),
-        vel_baseline_f("orbit.baseline_vel", Serializer<lin::Vector3d>(0,100000,100)),
-        prop_planner_state_f("prop.planner.state", Serializer<unsigned char>(2)),
+        pos_f("orbit.pos", Serializer<lin::Vector3d>(6771000,6971000, 28 + SerializerConstants::min_dvsz)),
```
[the remark this answers is not in the document]
Comment (kkrol27) [reply src/fsw/FCCode/FieldCreatorTask.hpp:28]: Do we have tickets open for these orbit state fields to look at the angular resolution as well?

**Comment by tanishqaggarwal** [comment]
I made a random list of who should look at which file. Could everyone take on at least the files they're listed for?

{‘ADCSBoxMonitor’: [‘Athena’, ‘Fatima’, ‘Andreea’, ‘Duncan’], ‘AttitudeEstimator’: [‘Tanishq’, ‘Maddy’], ‘PiksiControlTask/FieldCreatorTask’: [‘Shihao’, ‘Kirin’], ‘GomspaceController’: [‘Tanishq’, ‘Shihao’, ‘Kirin’, ‘Athena’], ‘MissionManager’: [‘Kyle’, ‘Athena’, ‘Fatima’, ‘Andreea’, ‘Duncan’], ‘PropController’: [‘Tanishq’, ‘Shihao’, ‘Maddy’, ‘Kyle’], ‘TimedControlTask’: [‘Kyle’]}

**Comment by tanishqaggarwal** [comment]
Here's a randomly generated set of verification assignments. I tried to balance this out so that newer folks review fewer files and experienced folks review more files.

{'ADCSBoxMonitor/AttitudeEstimator': ['Tanishq', 'Fatima', 'Andreea', 'Duncan'], 'PiksiControlTask/FieldCreatorTask': ['Maddy', 'Kirin', 'Kyle'], 'GomspaceController': ['Tanishq', 'Kirin', 'Kyle', 'Athena'], 'MissionManager': ['Shihao', 'Kyle', 'Athena', 'Andreea'], 'PropController': ['Tanishq', 'Shihao', 'Mahika'], 'TimedControlTask': ['Shihao', 'Athena', 'Fatima']}

Code:
```
import random

users = {"Tanishq" : 3, "Shihao": 3, "Maddy": 1, "Kirin": 2, "Mahika": 1, "Kyle": 3, "Athena": 3, "Fatima": 2, "Andreea": 2, "Duncan": 1}
executors = {
        "ADCSBoxMonitor/AttitudeEstimator" : ["Shihao", "Kyle", "Mahika"],
        "PiksiControlTask/FieldCreatorTask" : ["Tanishq", "Duncan"],
        "GomspaceController": ["Fatima", "Maddy"],
        "MissionManager": ["Tanishq","Maddy"],
        "PropController":["Andreea","Athena"],
        "TimedControlTask":["Tanishq", "Duncan"]
}

files = 3
verifiers = {}
for item in executors.keys():
    verifiers[item] = []

for user, load in users.items():
    user_files_added = 0
    while user_files_added < load:
        item = random.choice(list(executors.keys()))
        if not user in executors[item] and not user in verifiers[item]:
            verifiers[item].append(user)
            user_files_added += 1

print(verifiers)
```

File: src/fsw/FCCode/GomspaceController.cpp [hunk 9]
```diff
@@ -4,32 +4,34 @@ GomspaceController::GomspaceController(StateFieldRegistry &registry, unsigned in
...
     batt_threshold_f("gomspace.batt_threshold", batt_threshold_sr),
 
-    vboost_sr(0,4000,9), 
+    vboost_sr(0,8500,9), // see pg 11
     vboost1_f("gomspace.vboost.output1", vboost_sr),
     vboost2_f("gomspace.vboost.output2", vboost_sr),
     vboost3_f("gomspace.vboost.output3", vboost_sr),
 
-    vbatt_sr(5000,9000,10),
+    vbatt_sr(5000,9500,10), // see pg 9
```
Comment (knk38) [inline src/fsw/FCCode/GomspaceController.cpp:20]: why is the min not 0? 
Screen Shot 2020-11-15 at 10 08 25 AM

File: src/fsw/FCCode/GomspaceController.cpp [hunk 10]
```diff
@@ -4,32 +4,34 @@ GomspaceController::GomspaceController(StateFieldRegistry &registry, unsigned in
...
     vbatt_f("gomspace.vbatt", vbatt_sr),
 
-    curin_sr(0,10,10), 
+    curin_sr(0,10,16),
```
Comment (knk38) [inline src/fsw/FCCode/GomspaceController.cpp:23]: On page 9 of the data sheet, it lists current in as having a min of 0 A and max of 6 A

**Review by knk38 — COMMENTED** [review]
Looks good, was just confused on some of the gomspace points.

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 11]
```diff
@@ -8,23 +8,23 @@ ADCSBoxMonitor::ADCSBoxMonitor(StateFieldRegistry &registry,
     unsigned int offset, Devices::ADCS &_adcs)
     : TimedControlTask<void>(registry, "adcs_monitor", offset),
     adcs_system(_adcs),
-    rwa_speed_rd_sr(adcs::rwa::min_speed_read, adcs::rwa::max_speed_read, 16*3), //referenced from I2C_Interface.doc
+    rwa_speed_rd_sr(adcs::rwa::min_speed_read, adcs::rwa::max_speed_read, 8*3), //full res is 16*3
     rwa_speed_rd_f("adcs_monitor.rwa_speed_rd", rwa_speed_rd_sr),
-    rwa_torque_rd_sr(adcs::rwa::min_torque, adcs::rwa::max_torque, 16*3), //referenced from I2C_Interface.doc
+    rwa_torque_rd_sr(adcs::rwa::min_torque, adcs::rwa::max_torque, 8*3), //full res is 16*3
```
[the remark this answers is not in the document]
Comment (tanishqaggarwal) [reply src/fsw/FCCode/ADCSBoxMonitor.cpp:13]: Affirmative. They will get handled in #540

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 12]
```diff
@@ -8,23 +8,23 @@ ADCSBoxMonitor::ADCSBoxMonitor(StateFieldRegistry &registry,
...
     rwa_torque_rd_f("adcs_monitor.rwa_torque_rd", rwa_torque_rd_sr),
-    ssa_mode_rd(0,2,2), //referenced from Interface.doc
+    ssa_mode_rd(0,2,2), //3 modes
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/ADCSBoxMonitor.cpp:15]:
```suggestion
    ssa_mode_rd(2),
```

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 13]
```diff
@@ -8,23 +8,23 @@ ADCSBoxMonitor::ADCSBoxMonitor(StateFieldRegistry &registry,
...
     ssa_mode_f("adcs_monitor.ssa_mode", ssa_mode_rd),
-    ssa_vec_sr(-1,1,16*3), //referenced from I2C_Interface.doc
+    ssa_vec_sr(0,1,1 + SerializerConstants::min_fvsz), //full res is 16*3
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/ADCSBoxMonitor.cpp:17]:
```suggestion
    ssa_vec_sr(0,1,1 + SerializerConstants::min_fvsz),
```

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 14]
```diff
@@ -8,23 +8,23 @@ ADCSBoxMonitor::ADCSBoxMonitor(StateFieldRegistry &registry,
...
     ssa_vec_f("adcs_monitor.ssa_vec", ssa_vec_sr),
-    ssa_voltage_sr(adcs::ssa::min_voltage_rd, adcs::ssa::max_voltage_rd, 8),
+    ssa_voltage_sr(adcs::ssa::min_voltage_rd, adcs::ssa::max_voltage_rd, 8), //per voltage, full is 8
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/ADCSBoxMonitor.cpp:19]: What is the rationale for 8 bits?

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 15]
```diff
@@ -8,23 +8,23 @@ ADCSBoxMonitor::ADCSBoxMonitor(StateFieldRegistry &registry,
...
+    mag1_vec_sr(adcs::imu::min_mag1_rd_mag, adcs::imu::max_mag1_rd_mag, 8*3), //full res is 16*3
     mag1_vec_f("adcs_monitor.mag1_vec", mag1_vec_sr),
-    mag2_vec_sr(adcs::imu::min_mag2_rd_mag, adcs::imu::max_mag2_rd_mag, 16*3), //referenced from I2C_Interface.doc
+    mag2_vec_sr(adcs::imu::min_mag2_rd_mag, adcs::imu::max_mag2_rd_mag, 8*3), //full res is 16*3
     mag2_vec_f("adcs_monitor.mag2_vec", mag2_vec_sr),
-    gyr_vec_sr(adcs::imu::min_rd_omega, adcs::imu::max_rd_omega, 16*3), //referenced from I2C_Interface.doc
+    gyr_vec_sr(adcs::imu::min_rd_omega, adcs::imu::max_rd_omega, 16*3), //full res is 16*3
     gyr_vec_f("adcs_monitor.gyr_vec", gyr_vec_sr),
-    gyr_temp_sr(adcs::ssa::min_voltage_rd, adcs::ssa::max_voltage_rd, 16), //referenced from I2C_Interface.doc
+    gyr_temp_sr(adcs::ssa::min_voltage_rd, adcs::ssa::max_voltage_rd, 8), //full res is 16
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/ADCSBoxMonitor.cpp:27]: The bounds here are for SSA voltage, not gyro temperature. @kkrol27 @shihaocao

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 16]
```diff
@@ -8,23 +8,23 @@ ADCSBoxMonitor::ADCSBoxMonitor(StateFieldRegistry &registry,
...
+    ssa_voltage_sr(adcs::ssa::min_voltage_rd, adcs::ssa::max_voltage_rd, 8), //per voltage, full is 8
     ssa_voltages_f(),
-    mag1_vec_sr(adcs::imu::min_mag1_rd_mag, adcs::imu::max_mag1_rd_mag, 16*3), //referenced from I2C_Interface.doc
+    mag1_vec_sr(adcs::imu::min_mag1_rd_mag, adcs::imu::max_mag1_rd_mag, 8*3), //full res is 16*3
     mag1_vec_f("adcs_monitor.mag1_vec", mag1_vec_sr),
-    mag2_vec_sr(adcs::imu::min_mag2_rd_mag, adcs::imu::max_mag2_rd_mag, 16*3), //referenced from I2C_Interface.doc
+    mag2_vec_sr(adcs::imu::min_mag2_rd_mag, adcs::imu::max_mag2_rd_mag, 8*3), //full res is 16*3
     mag2_vec_f("adcs_monitor.mag2_vec", mag2_vec_sr),
-    gyr_vec_sr(adcs::imu::min_rd_omega, adcs::imu::max_rd_omega, 16*3), //referenced from I2C_Interface.doc
+    gyr_vec_sr(adcs::imu::min_rd_omega, adcs::imu::max_rd_omega, 16*3), //full res is 16*3
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/ADCSBoxMonitor.cpp:25]: Note to other reviewers: ignore these serializers. They are being addressed in #540

File: src/fsw/FCCode/AttitudeEstimator.cpp [hunk 17]
```diff
@@ -20,8 +22,9 @@ AttitudeEstimator::AttitudeEstimator(StateFieldRegistry &registry,
     mag_flag_f("attitude_estimator.mag_flag", Serializer<bool>()),
     b_body_f("attitude_estimator.b_body"),
     q_body_eci_est_f("attitude_estimator.q_body_eci", Serializer<lin::Vector4f>()),
-    w_body_est_f("attitude_estimator.w_body", Serializer<lin::Vector3f>(-55, 55, 32*3)),
-    fro_P_est_f("attitude_estimator.fro_P", Serializer<float>(0.0, 0.1, 16))
+    w_body_est_f("attitude_estimator.w_body", 
+    Serializer<lin::Vector3f>(adcs::imu::min_rd_omega, adcs::imu::max_rd_omega, 16*3)), //full res is 32*3;
+    fro_P_est_f("attitude_estimator.fro_P", Serializer<float>(0.0, 0.1, 32)) //max res 32
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/AttitudeEstimator.cpp:27]: What is the rationale for the bounds being 0.0 to 0.1?

File: src/fsw/FCCode/FieldCreatorTask.hpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (tanishqaggarwal) [reply src/fsw/FCCode/FieldCreatorTask.hpp:28]: Now we do (#552)

File: src/fsw/FCCode/MissionManager.cpp [hunk 18]
```diff
@@ -16,21 +16,19 @@ const constexpr std::array<mission_state_t, 5> MissionManager::fault_responsive_
...
+                                                                                    sat_designation_f("pan.sat_designation", Serializer<unsigned char>(3), 100),
```
Comment (afoarce) [inline src/fsw/FCCode/MissionManager.cpp:30]: What is the rationale for this change?

Comment (tanishqaggarwal) [reply src/fsw/FCCode/MissionManager.cpp:30]: For `deployment.elapsed`: this is not a field that would change after the deployment period; there's no need to downlink it, so no need to set good serialization parameters for it.

For `pan.sat_designation`: this field has three possible values (undecided, leader, follower), so oops this actually has values from 0-2 and the original value was correct. Nice catch

File: src/fsw/FCCode/PropController.cpp [hunk 19]
```diff
@@ -11,27 +11,27 @@
...
+      ctrl_cycles_per_filling_period("prop.ctrl_cycles_per_filling", Serializer<unsigned int>(25)),
+      ctrl_cycles_per_cooling_period("prop.ctrl_cycles_per_cooling", Serializer<unsigned int>(100)),
       tank1_valve("prop.tank1.valve_choice", Serializer<unsigned int>(1)),
-
-      tank2_pressure_f("prop.tank2.pressure", Serializer<float>(0, 150, 4)),
-      tank2_temp_f("prop.tank2.temp", Serializer<float>(-200, 200, 4)),
-      tank1_temp_f("prop.tank1.temp", Serializer<float>(-200, 200, 4)),
+      tank2_pressure_f("prop.tank2.pressure", Serializer<float>(0, 150, 8)),
+      // https://docs.google.com/spreadsheets/d/11-WSDgYckQGl1wP8uMO4zKPSDAdWwGQj/edit#gid=422064948
+      tank2_temp_f("prop.tank2.temp", Serializer<float>(-60, 155, 8)),
```
Comment (18kudlugim) [inline src/fsw/FCCode/PropController.cpp:33]: What's the rationale behind using 8 bytes instead of 4 for tank temp?

File: src/fsw/FCCode/PropController.cpp [hunk 20]
```diff
@@ -11,27 +11,27 @@
...
-      cycles_until_firing("prop.cycles_until_firing", Serializer<unsigned int>(256)),
+      prop_state_f("prop.state", Serializer<unsigned int>(9)),
+      cycles_until_firing("prop.cycles_until_firing", Serializer<unsigned int>(orbit_ccno)),
       sched_valve1_f("prop.sched_valve1", Serializer<unsigned int>(999)),
       sched_valve2_f("prop.sched_valve2", Serializer<unsigned int>(999)),
       sched_valve3_f("prop.sched_valve3", Serializer<unsigned int>(999)),
       sched_valve4_f("prop.sched_valve4", Serializer<unsigned int>(999)),
-      sched_intertank1_f("prop.sched_intertank1", Serializer<unsigned int>(999 * 1000)),
-      sched_intertank2_f("prop.sched_intertank2", Serializer<unsigned int>(999 * 1000)),
+      sched_intertank1_f("prop.sched_intertank1", Serializer<unsigned int>(2000)),
```
[the remark this answers is not in the document]
Comment (athena255) [reply src/fsw/FCCode/PropController.cpp:20]: Because prop schedules are in units of milliseconds an we don't want 999 seconds, we want 2 seconds.

File: src/fsw/FCCode/GomspaceController.cpp [hunk 21]
```diff
@@ -4,32 +4,34 @@ GomspaceController::GomspaceController(StateFieldRegistry &registry, unsigned in
...
     vboost1_f("gomspace.vboost.output1", vboost_sr),
     vboost2_f("gomspace.vboost.output2", vboost_sr),
     vboost3_f("gomspace.vboost.output3", vboost_sr),
 
-    vbatt_sr(5000,9000,10),
+    vbatt_sr(5000,9500,10), // see pg 9
     vbatt_f("gomspace.vbatt", vbatt_sr),
 
-    curin_sr(0,10,10), 
+    curin_sr(0,6,10),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/GomspaceController.cpp:23]:
```suggestion
    curin_sr(0,6000,10), 
```

Comment (tanishqaggarwal) [inline src/fsw/FCCode/GomspaceController.cpp:23]:
```suggestion
    curin_sr(0,6000,10), 
```

File: src/fsw/FCCode/GomspaceController.cpp [hunk 22]
```diff
@@ -4,32 +4,34 @@ GomspaceController::GomspaceController(StateFieldRegistry &registry, unsigned in
...
     vbatt_f("gomspace.vbatt", vbatt_sr),
 
-    curin_sr(0,10,10), 
+    curin_sr(0,6,10), 
     curin1_f("gomspace.curin.output1", curin_sr),
     curin2_f("gomspace.curin.output2", curin_sr),
     curin3_f("gomspace.curin.output3", curin_sr),
 
-    cursun_sr(0,10,10), 
+    cursun_sr(0,3,10),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/GomspaceController.cpp:28]:
```suggestion
    cursun_sr(0,3000,10), 
```

File: src/fsw/FCCode/GomspaceController.cpp [hunk 23]
```diff
@@ -4,32 +4,34 @@ GomspaceController::GomspaceController(StateFieldRegistry &registry, unsigned in
...
     curin1_f("gomspace.curin.output1", curin_sr),
     curin2_f("gomspace.curin.output2", curin_sr),
     curin3_f("gomspace.curin.output3", curin_sr),
 
-    cursun_sr(0,10,10), 
+    cursun_sr(0,3,10), 
     cursun_f("gomspace.cursun", cursun_sr),
 
-    cursys_sr(0,10,10), 
+    cursys_sr(0,12,10),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/GomspaceController.cpp:31]:
```suggestion
    cursys_sr(0,12000,10), 
```

File: src/fsw/FCCode/GomspaceController.cpp [hunk 24]
```diff
@@ -4,32 +4,34 @@ GomspaceController::GomspaceController(StateFieldRegistry &registry, unsigned in
...
-    cursun_sr(0,10,10), 
+    cursun_sr(0,3,10), 
     cursun_f("gomspace.cursun", cursun_sr),
 
-    cursys_sr(0,10,10), 
+    cursys_sr(0,12,10), 
     cursys_f("gomspace.cursys", cursys_sr),
 
-    curout_sr(0,10,10), 
+    curout_sr(0,12,10),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/GomspaceController.cpp:34]:
```suggestion
    curout_sr(0,12000,10), 
```

File: src/fsw/FCCode/PropController.cpp [hunk 25]
```diff
@@ -11,27 +11,27 @@
...
       max_venting_cycles("prop.max_venting_cycles", Serializer<unsigned int>(50)),
       ctrl_cycles_per_close_period("prop.ctrl_cycles_per_closing", Serializer<unsigned int>(50)),
 
       max_pressurizing_cycles("prop.max_pressurizing_cycles", Serializer<unsigned int>(50)),
-      threshold_firing_pressure("prop.threshold_firing_pressure", Serializer<float>(10, 50, 4)),
-      ctrl_cycles_per_filling_period("prop.ctrl_cycles_per_filling", Serializer<unsigned int>(50)),
-      ctrl_cycles_per_cooling_period("prop.ctrl_cycles_per_cooling", Serializer<unsigned int>(50)),
+      threshold_firing_pressure("prop.threshold_firing_pressure", Serializer<float>(10, 50, 6)),
+      ctrl_cycles_per_filling_period("prop.ctrl_cycles_per_filling", Serializer<unsigned int>(25)),
+      ctrl_cycles_per_cooling_period("prop.ctrl_cycles_per_cooling", Serializer<unsigned int>(100)),
```
[the remark this answers is not in the document]
Comment (athena255) [reply src/fsw/FCCode/PropController.cpp:29]: prop.threshold_firing_pressure is (10, 50, 4) but should be changed to (10, 50, 6)
prop.ctrl_cycles_per_filling is kind of high because this is the number of control cycles in 1 second (which is ~5 for a control cycle time of 170 ms)
prop.ctrl_cycles_per_cooling is too low because this is the number of control cycles in 10 seconds, which is (10*1000/170 ) = 58.8

File: src/fsw/FCCode/MissionManager.cpp (hunk 18 above)
Comment (tanishqaggarwal) [reply src/fsw/FCCode/MissionManager.cpp]: Discussed offline and resolved.

File: src/fsw/FCCode/AttitudeEstimator.cpp (hunk 17 above)
Comment (kkrol27) [reply src/fsw/FCCode/AttitudeEstimator.cpp:27]: #541

Comment (kkrol27) [reply src/fsw/FCCode/AttitudeEstimator.cpp:27]: There is no rational atm

File: src/fsw/FCCode/PropController.cpp (hunk 19 above)
Comment (athena255) [reply src/fsw/FCCode/PropController.cpp:33]: prop.tank2.pressure was (0, 150, 4), which gives intervals of 9.375, which is too high. 8 bytes would allow us to have 1 degree intervals.
prop.tank2.temp and prop.tank1.temp resolution was definitely too low because values range between -200 and 200

File: src/fsw/FCCode/ADCSBoxMonitor.cpp (hunk 14 above)
Comment (tanishqaggarwal) [reply src/fsw/FCCode/ADCSBoxMonitor.cpp:19]: Discussed offline. Going to change this back to 12 bits because (a) it's not really stuff that we downlink unless something is wrong, so for the most part changing this value doesn't affect telemetry, (b) to enable doing "science" with the values if we so choose.

```suggestion
    ssa_voltage_sr(adcs::ssa::min_voltage_rd, adcs::ssa::max_voltage_rd, 12), // Full resolution to enable scientific analysis of the sun sensing algorithm
```

File: src/fsw/FCCode/TimedControlTask.hpp (hunk 7 above)
Comment (athena255) [reply src/fsw/FCCode/TimedControlTask.hpp:182]: Shouldn't this be a function of the PAN::control_cycle_time_us?

Comment (tanishqaggarwal) [reply src/fsw/FCCode/TimedControlTask.hpp:182]: Agreed offline to the following: 2^18 ~ 256,000 so unless our control cycle time exceeds 256 ms we will still have 1 microsecond accuracy. The control cycle being this large would be very bad news for our GNC algorithms' performances, I don't think it's worth complicating the bitsize calculation and make it a function of control cycle time.

File: src/fsw/FCCode/PropController.cpp (hunk 25 above)
[the remark this answers is not in the document]
Comment (tanishqaggarwal) [reply src/fsw/FCCode/PropController.cpp:29]: Makes sense to me

File: src/fsw/FCCode/ADCSBoxMonitor.cpp (hunk 15 above)
Comment (shihaocao) [reply src/fsw/FCCode/ADCSBoxMonitor.cpp:27]: great catch thanks

File: src/fsw/FCCode/MissionManager.cpp (hunk 1 above)
Comment (shihaocao) [reply src/fsw/FCCode/MissionManager.cpp:19]: @kkrol27 I think our detumble safety factor * momentum needs to be under the momentum that an impulse of a single thruster firing, some quick math might determine that the detumble safety factor should never be as high as 1, and thus we might want to consider if we want lower bounds as 1% resolution might not be good enough if the safety factor is like 0.002?

Just writing this down for thought / documentation

Comment (shihaocao) [reply src/fsw/FCCode/MissionManager.cpp:19]: #561

File: src/fsw/FCCode/MissionManager.cpp (hunk 5 above)
Comment (shihaocao) [reply src/fsw/FCCode/MissionManager.cpp:29]: We should still set it to something reasonable right? If there's a bug in the code that somehow downlinks this, it would wreak havoc if it was unreasonable?

Comment (shihaocao) [reply src/fsw/FCCode/MissionManager.cpp:29]: Oops 15000 is max not number of bits, yeah this serializer seems chill

File: src/fsw/FCCode/MissionManager.cpp (hunk 4 above)
Comment (shihaocao) [reply src/fsw/FCCode/MissionManager.cpp:23]: I feel like we'd never realistically wait in docking for more than say 4 hours right? If we're a few meters apart it should only take a few minutes for the magnets to suck the satellites together?

File: src/fsw/FCCode/MissionManager.cpp (hunk 1 above)
Comment (tanishqaggarwal) [reply src/fsw/FCCode/MissionManager.cpp:19]: Ok let's track changing this constant on that ticket, I'll resolve this comment so that this PR can move forward

File: src/fsw/FCCode/MissionManager.cpp (hunk 4 above)
Comment (tanishqaggarwal) [reply src/fsw/FCCode/MissionManager.cpp:23]: If it might only take a few minutes, might as well just give an hour to make docking extra-certainly not dependent on this timeout. This is a scenario where we basically get compression for free by being more relaxed about our mission constraints.

File: src/fsw/FCCode/PropController.cpp (hunk 20 above)
[the remark this answers is not in the document]
Comment (tanishqaggarwal) [reply src/fsw/FCCode/PropController.cpp:20]: Makes sense to me

**Review by afoarce — APPROVED** [review]

**Review by kkrol27 — APPROVED** [review]
Looks good to me!

**Review by Duncan-McD — APPROVED** [review]

File: src/fsw/FCCode/MissionManager.cpp (hunk 4 above)
Comment (shihaocao) [reply src/fsw/FCCode/MissionManager.cpp:23]: Okay fair enough.

File: src/fsw/FCCode/MissionManager.cpp (hunk 5 above)
Comment (shihaocao) [reply src/fsw/FCCode/MissionManager.cpp:29]: I'm not sure what's going with the eeprom_save_duration?

Comment (tanishqaggarwal) [reply src/fsw/FCCode/MissionManager.cpp:29]: The elapsed deployment time is saved once every 500 control cycles (~once every minute) so that if the satellite shuts down during the 30-minute deployment wait, it's not going to restart the timer over and over

**Review by madeleinemc — APPROVED** [review]

**Review by athena255 — APPROVED** [review]

**Review by athena255 — APPROVED** [review]

**Review by knk38 — APPROVED** [review]
Yup i see the gomspace fields are changed now so everything looks good to me

**Review by 18kudlugim — APPROVED** [review]
Looks good!
